### PR TITLE
Add documentation for the 'Create Type' statement

### PR DESCRIPTION
**Pull request by Loreith**
Disclaimer that I learned all of this from the discord (thanks Elliana and __Alex__!) and trial and error, not from actually reading the source or a grammar file.

Also please feel free to cut down the railroad diagram or examples if you feel like I went a little overboard with it - I just noticed that there wasn't any mention of being able to create structs or unions elsewhere and wanted to be specific about them since the syntax is a little more involved than "just write the type identifier".

### CODE REVIEW

File: docs/sql/statements/create_type.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+```
+
+### Syntax
+
+<div id="rrdiagram"></div>
+
+`CREATE TYPE` defines a new data type available to this duckdb instance. These new
+types can then be inspected in the `duckdb_types` table.
+
+For fully custom operators and things on a custom type, you'll have to go into the
```
Comment (Mause) [inline docs/sql/statements/create_type.md:30]: Replace "things"?

File: docs/sql/statements/create_type.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+### Examples
+
+```sql
+-- create a simple enum type
+CREATE TYPE mood AS ENUM ('happy', 'sad', 'curious');
+-- create a simple struct type
+CREATE TYPE many_things AS STRUCT(k INTEGER, l VARCHAR);
+-- create a simple union type
+CREATE TYPE one_thing AS UNION(numer INTEGER, string VARCHAR);
```
Comment (Mause) [inline docs/sql/statements/create_type.md:18]:
```suggestion
CREATE TYPE one_thing AS UNION(number INTEGER, string VARCHAR);
```
?

**Review by Mause — CHANGES_REQUESTED** [review]

File: docs/sql/statements/create_type.md (hunk 2 above)
Comment (Loreith) [reply docs/sql/statements/create_type.md]: well done me:p

**Comment by Mause** [comment]
Thanks!

(Also hi from another Greens person, from Boorloo!)

**Pull request merged**